src/commands/readme.ts

```diff
@@ -141,13 +145,27 @@ USAGE
 
   renderCommand(config: Config.IConfig, c: Config.Command): string {
     this.debug('rendering command', c.id)
-    const title = template({config, command: c})(c.description || '').trim().split('\n')[0]
-    const help = new Help(config, {stripAnsi: true, maxWidth: columns})
+    const HelpClass = getHelpClass(config)
+    const help = new HelpClass(config, {stripAnsi: true, maxWidth: columns})
+
     const header = () => `## \`${config.bin} ${this.commandUsage(config, c)}\``
+
+    const generateCommandHelp = () =>  {
+      const original = process.stdout.write
```

Comment:
I don't love this. Can we make the help class more flexible to return the command help instead? e.g `help.formatCommandHelp` and the showCommandHelp would call that to pass to stdout.

Reply:
We can, but it would mean a breaking change to `@oclif/plugin-help` because we currently define the minimum contract to be just `showHelp` and `showCommandHelp` and [in the docs we advise users to write directly to stdout](https://oclif.io/docs/help_classes#extending-the-helpbase-class).

Reply:
The base help class already has format methods. It seems like a natural extension to make it part of the interface. It wouldn't really be a breaking change. It would just mean that if people want to take advantage of the new customer help in the readme they would have to implement the additional methods in the interface. 

Also, we should still advise writing directly to stdout in the show methods, but update the docs to utilize the format methods for additional abstraction and customization. 

With that said, I don't feel strongly enough to fight for it. I just don't like this pattern.

Reply:
> if people want to take advantage of the new customer help in the readme they would have to implement the additional methods in the interface

I think this might be the way to go. Perhaps something like:

```sh
$ npm run oclif-dev readme
 ›   Error: You are using a custom help class without required format methods.
 ›   Please define `formatCommand` etc. 
 ›   Please see https://oclif.io/docs/help_classes for more.
```

Reply:
I’ve addressed this in 57b01c917e6e7fb2bf5cb834c327977254af72c0.

With the new approach (which is close in spirit to @chadian’s original approach) the readme generator will try to use `formatCommand`, then `command`, then throw a helpful error if there’s nothing usable on the custom help class.

The error looks like this:

<img width="1046" alt="Screen Shot 2020-11-05 at 14 08 25" src="https://user-images.githubusercontent.com/34030/98244911-75516980-1f70-11eb-8a3c-3ef6f30317a8.png">